broker/bam/src/configuration/applier/state.cc

```diff
@@ -252,15 +253,11 @@ void applier::state::save_to_cache(persistent_cache& cache) {
 void applier::state::load_from_cache(persistent_cache& cache) {
   _logger->debug("BAM: Loading restoring inherited downtimes and BA states");
 
-  std::shared_ptr<io::data> d;
-  cache.get(d);
+  std::shared_ptr<io::data> data;
+  cache.get(data);
+  std::shared_ptr<io::data> d = neb::bbdo2_to_bbdo3(data);
   while (d) {
     switch (d->type()) {
-      case inherited_downtime::static_type(): {
-        const inherited_downtime& dwn =
-            *std::static_pointer_cast<const inherited_downtime>(d);
-        _ba_applier.apply_inherited_downtime(dwn);
-      } break;
       case pb_inherited_downtime::static_type(): {
```

Comment:
Why not just translate when needed?
you always call bbdo2_to_bbdo3 but you could call it only on old events.